src/coreComponents/constitutive/solid/CoupledSolid.hpp

```diff
@@ -0,0 +1,188 @@
+  {
+    return m_solidUpdate;
+  }
+
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  typename PORO_TYPE::KernelWrapper const getPorosityModel() const
+  {
+    return m_porosityUpdate;
+  }
+
+protected:
+  typename SOLID_TYPE::KernelWrapper const m_solidUpdate;
+
+  typename PORO_TYPE::KernelWrapper const m_porosityUpdate;
+};
+
+
+
+/**
+ * @brief Class to represent a coupled solid model
+ */
+template< typename SOLID_TYPE,
+          typename PORO_TYPE >
+class CoupledSolid : public ConstitutiveBase
+{
+public:
+
+  /**
+   * @brief Constructor
+   * @param name Object name
+   * @param parent Object's parent group
+   */
+  CoupledSolid( string const & name, dataRepository::Group * const parent );
+
+  /// Destructor
+  virtual ~CoupledSolid() override;
+
+  /**
+   * @brief Catalog name
+   * @return Static catalog string
+   */
+  static string catalogName() { return string( "Poro" ) + SOLID_TYPE::m_catalogNameString; }
+
+  /**
+   * @brief Get catalog name
+   * @return Catalog name string
+   */
+  virtual string getCatalogName() const override { return catalogName(); }
+
+  /// Post-process XML input
+  virtual void postProcessInput() override;
+
+  struct viewKeyStruct
+  {
+    static constexpr char const * solidModelNameString() { return "solidModelName"; }
+    static constexpr char const * porosityModelNameString() { return "porosityModelName"; }
+  };
+
+  /**
+   * @brief Create a instantiation of the PorousSolidUpdates class
+   *        that refers to the data in this.
+   * @return An instantiation of PorousSolidUpdates.
+   */
+  CoupledSolidUpdates< SOLID_TYPE, PORO_TYPE > createKernelUpdates() const
+  {
+
+    return CoupledSolidUpdates< SOLID_TYPE, PORO_TYPE >( m_solidModel,
+                                                         m_porosityModel );
+  }
+
+protected:
+
+  // the solid model
+  SOLID_TYPE * m_solidModel;
+
+  // the porosity model
+  PORO_TYPE * m_porosityModel;
+
+  // the name of the solid model
+  string m_solidModelName;
+
+  // the name of the porosity model
+  string m_porosityModelName;
+
+
+  // PERMEABILITY_TYPE * m_permModel;
+};
+
+
+template< typename SOLID_TYPE,
+          typename PORO_TYPE >
+CoupledSolid< SOLID_TYPE, PORO_TYPE >::CoupledSolid( string const & name, Group * const parent ):
+  ConstitutiveBase( name, parent ),
+  m_solidModel( nullptr ),
+  m_porosityModel( nullptr ),
+  m_solidModelName(),
+  m_porosityModelName()
+{
+  registerWrapper( viewKeyStruct::solidModelNameString(), &m_solidModelName ).
+    setInputFlag( dataRepository::InputFlags::REQUIRED ).
+    setDescription( "Name of the solid model." );
+
+  registerWrapper( viewKeyStruct::porosityModelNameString(), &m_porosityModelName ).
+    setInputFlag( dataRepository::InputFlags::REQUIRED ).
+    setDescription( "Name of the porosity model." );
+}
+
+template< typename SOLID_TYPE,
+          typename PORO_TYPE >
+CoupledSolid< SOLID_TYPE, PORO_TYPE >::~CoupledSolid()
+{}
```

Comment:
Sure, although it would be interesting to know why it was done for a lot of the classes in geosx. If we should always use default then we should probably to a proper clean up.

Reply:
No real reason & I'm replacing these with `= default` for fluid models. Most models can also place `= default` directly in the class declaration.